python/perf-kernels/flash-attention.py

```diff
@@ -996,6 +1058,13 @@ def test_op_fwd(Z, H, N_CTX_Q, N_CTX_K, D_HEAD, causal, dtype=torch.float16):
         mask = torch.tril(torch.ones(N_CTX_Q, N_CTX_K, device="cuda"), 
                           diagonal=N_CTX_K-N_CTX_Q)
         scores[:, :, mask==0] = float("-inf")
+    if use_alibi:
+        q_idx = torch.arange(N_CTX_Q, dtype=torch.int32, device="cuda").unsqueeze(-1)
+        k_idx = torch.arange(N_CTX_K, dtype=torch.int32, device="cuda").unsqueeze(0)
+        relative_pos = torch.tril(k_idx - q_idx)
+        alibi = alibi_slopes.unsqueeze(-1).unsqueeze(-1) * relative_pos # (Z, H, N_CTX_Q, N_CTX_K)
+        scores+= alibi
```

Comment:
Hmm so the documentation on Tri Dao's implementation says 

```
alibi_slopes: (nheads,) or (batch_size, nheads), fp32. A bias of
            (-alibi_slope * |i + seqlen_k - seqlen_q - j|)
            is added to the attention score of query i and key j.
```

So the value it must be multiplied by would be `torch.abs(q_idx - k_idx + K_CTX - Q_CTX)`. But here instead of abs it is `max(0, q_idx - k_idx)` from what I can tell. I'm probably missing something?

Reply:
thanks, I fixed it.